bundles/org.eclipse.jface/src/org/eclipse/jface/resource/DeviceResourceDescriptor.java

```diff
@@ -25,11 +25,13 @@
  * two equal descriptors at hand, e.g. decorating an identical object.
  * </p>
  *
+ * @param <R> The resource's type described by this descriptor
+ *
  * @see org.eclipse.jface.resource.ResourceManager
  *
  * @since 3.1
  */
-public abstract class DeviceResourceDescriptor {
+public abstract class DeviceResourceDescriptor<R> {
```

Comment:
If this class would be created new, we could consider to use `R extends org.eclipse.swt.graphics.Resource` but this would likely break binary compatibility because the erasure would be `Resource` instead of `Object`.